rasa_sdk/knowledge_base/actions.py

```diff
@@ -182,9 +195,8 @@ async def _query_objects(
         )
 
         last_object = None if len(objects) > 1 else objects[0][key_attribute]
-
         slots = [
-            SlotSet(SLOT_OBJECT_TYPE, object_type),
+            SlotSet(SLOT_OBJECT_TYPE, None),
```

Comment:
Why is `object_type` replaced with `None` here?

Reply:
In rasa’s current implementation, the object_type value is set to the slot_object_type. This places a requirement for the user to ask the bot to first list a set of objects of an object type, for e.g: ‘list me some restaurants’, before a user can ask ‘what is the price range of Berlin Burrito Company?’. If the user directly asks ‘what is the price range of Berlin Burrito Company?’, the bot detects that there is no object_type and asks the user to rephrase. This is the problem we are trying to solve.

In our proposed solution, there can be instances that the object type has to be extracted while the action is executed. This means we need to reset the object_type slot after every run. This enables the user to hold a conversation similar to the example below.
**E.g:
```
Your input -> what is the price range of Berlin Burrito Company ?
‘Berlin Burrito Company’ has the value ‘cheap’ for attribute ‘price-range’.
Your input -> what is the star rating of B&B
‘B&B (Berlin)’ has the value ‘1’ for attribute ‘star-rating’.
```**
In the above example, Berlin Burrito Company is a restaurant and B&B is a hotel. By resetting the `object_type slot`  throughout the conversation, the current solution can answer inquiries in this manner.

Reply:
Ok, understood now. Could you please add a comment with a brief explanation of why this resetting is required.
Also a question I had is if you've tested when a user asks directly about an object that is not in the knowledge base (let's say a restaurant that is missing from the knowledge base), how is this handled?